src/pkg/retention/launcher.go

```diff
@@ -75,11 +71,13 @@ type Launcher interface {
 
 // NewLauncher returns an instance of Launcher
 func NewLauncher(projectMgr project.Manager, repositoryMgr repository.Manager,
-	retentionMgr Manager) Launcher {
+	retentionMgr Manager, execMgr task.ExecutionManager, taskMgr task.Manager) Launcher {
```

Comment:
do we still need the launcher here? Seems that the methods can be merged into the controller.

Reply:
this retention's launcher still have some special logic not included in task manager